### ISSUE TITLE
expression.simplify with weight substitution does not work

### ISSUE THREAD

**Issue by hejung** [darioizzo/dcgp#48]
Setup: dcgpy v1.2.1 from PyPI installed via pip

The following code will throw a `TypeError`. I believe the reason is that `expression.get_arity()` now returns a list with arities per node instead of a single int.
```python
from dcgpy import kernel_set_gdual_vdouble as kernel_set
from dcgpy import expression_weighted_gdual_vdouble as dcgpy_expression

kernels = ['sum', 'diff']
n_in = 1
n_out = 1
rows = 1
cols = 15
levels_back = 16
arity = 2
kernels = kernel_set(kernels)()
ex = dcgpy_expression(n_in, n_out, rows=rows, cols=cols,
                      levels_back=levels_back, arity=arity,
                      kernels=kernels)
ex.simplify(['x'], subs_weights=True)
```

**Comment by darioizzo** [comment]
What line throws the error? Also can you post the exact output?

**Comment by hejung** [comment]
The last line in the example, `ex.simplify(['x'], subs_weights=True)` throws the error:
```
../python3.7/site-packages/dcgpy/_simplify.py in _sympy_simplify(self, in_sym, subs_weights, erc)
     65         a = self.get_arity()
     66         for i in range(r * c):
---> 67             for j in range(a):
     68                 ws = 'w' + str(n + i) + '_' + str(j)
     69                 ns[ws] = sympy.Symbol(ws, real = True)

TypeError: 'list' object cannot be interpreted as an integer
```
This happens only with `subs_weights=True`.

**Comment by darioizzo** [comment]
Thanks, you are right its a consequence of the update to the vecotial arity .... its a very simple thing to fix and will do in the next upcoming release ....

**Comment by hejung** [comment]
Great! Thank you for the fast reply!
For now I will just stay with dcgpy v1.1 from PyPi where everything works as expected.

As a side note, just to let you know:
If installing dcgpy v1.1 using pip, i.e. with `pip install dcgpy==1.1` I ran into another error which seems to be related to the pyaudi version.
 Using `expression.set_weight(j, k, gdual_vdouble([1.]))`  for an `expression_weighted_gdual_vdouble` results in a type mismatch error.
However, I was able to avoid the error by manually installing pyaudi  v1.6 (instead of v1.6.5 which is what pip gets when I do  a `pip install dcgpy==1.1`).
